UnitTests/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachineTests.swift

```diff
@@ -0,0 +1,337 @@
+//
+//  ContextualOnboardingStateMachineTests.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import PrivacyDashboard
+@testable import DuckDuckGo_Privacy_Browser
+
+@available(macOS 12.0, *)
+class ContextualOnboardingStateMachineTests: XCTestCase {
+
+    var stateMachine: ContextualOnboardingStateMachine!
+    var mockTrackerMessageProvider: MockTrackerMessageProvider!
+    var tab: Tab!
+    let expectation = XCTestExpectation()
+    var window: NSWindow!
+    var tabViewModel: TabViewModel {
+        (window.contentViewController as! MainViewController).browserTabViewController.tabViewModel!
+    }
+
+    @MainActor override func setUp() {
+        super.setUp()
+        WebTrackingProtectionPreferences.shared.isGPCEnabled = false
+
+        window = WindowsManager.openNewWindow(with: .none)!
+        UserDefaultsWrapper<Any>.clearAll()
+        mockTrackerMessageProvider = MockTrackerMessageProvider(expectation: expectation)
+        stateMachine = ContextualOnboardingStateMachine(trackerMessageProvider: mockTrackerMessageProvider)
+        tab = Tab(url: URL.duckDuckGo)
+    }
+
+    override func tearDown() {
+        stateMachine = nil
+        mockTrackerMessageProvider = nil
+        tab = nil
+        window.close()
+        window = nil
+
+        WebTrackingProtectionPreferences.shared.isGPCEnabled = true
```

Comment:
out of curiosity, what does this `WebTrackingProtectionPreferences.shared.isGPCEnabled` do?

Reply:
Ah it should not be there I was experimenting with stuff and it was left over!